site-service/src/main/java/uk/ac/ox/ndph/mts/site_service/service/SiteServiceImpl.java

```diff
@@ -116,11 +143,16 @@ public String save(final Site site) {
      * @throws InvariantException if the list from the store is empty
      */
     @Override
-    public List<Site> findSites() {
-        final List<Site> sites = this.siteStore.findAll();
+    public List<SiteDTO> findSites() {
+        final List<SiteDTO> sites = this.siteStore.findAll().stream()
+            .map(site -> convertSite(site))
+            .collect(Collectors.toList());
         if (sites.isEmpty()) {
             throw new InvariantException(Services.NO_ROOT_SITE.message());
         }
+
+        populateParentName(sites);
```

Comment:
This is the performance improvement that saves the UI from having to come back with n+1 little requests for parentnames?  I think this is a good idea.